src/Polkadot/ScaleCodec.h

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "../BinaryCoding.h"
+#include "../Data.h"
+#include "../SS58Address.h"
+#include "../PublicKey.h"
+#include <boost/multiprecision/cpp_int.hpp>
+
+/// Reference https://github.com/soramitsu/kagome/blob/master/core/scale/scale_encoder_stream.cpp
+using CompactInteger = boost::multiprecision::cpp_int;
+
+namespace TW::Polkadot {
+
+static constexpr size_t kMinUint16 = (1ul << 6u);
+static constexpr size_t kMinUint32 = (1ul << 14u);
+static constexpr size_t kMinBigInteger = (1ul << 30u);
+
+inline size_t countBytes(CompactInteger value) {
+    if (0 == value) {
+        return 1;
+    }
+
+    size_t size = 0;
+    while (value > 0) {
+        ++size;
+        value >>= 8;
+    }
+
+    return size;
+}
+
+inline Data encodeCompact(CompactInteger value) {
+    auto data = Data{};
+
+    if (value < kMinUint16) {
+        auto v = value.convert_to<uint8_t>() << 2u;
+        data.push_back(static_cast<uint8_t>(v));
+        return data;
+    } else if (value < kMinUint32) {
+        auto v = (value.convert_to<uint16_t>() << 2u);
+        v += 0b01; // set 0b01 flag
```

Comment:
Since other values are hex, I would use 0x01, which is the same, but more readable here.

Reply:
done